### PR TITLE
Add initial vale configuration and apply editorial changes to documentation

### LINKED ISSUE #3566: Integrate Vale for linting docs according to editorial style guide
## Description

We can integrate Vale (https://github.com/errata-ai/vale) into the pre-commit hooks to ensure that our documentation has a consistent style guide that can be enforced. Vale has a significant number of stars on GitHub, seems actively maintained and [supports pre-commit](https://github.com/errata-ai/vale/blob/v2/.pre-commit-hooks.yaml).

This is related to WordPress/openverse#368.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #3566 by @dhruvkb 

## Description

I've picked this work up from @ngken0995 so that Ken can focus on other PRs.

When we originally started looking into Vale, we were misinformed about how it worked. We thought that you had to commit the styles files to the repository. However, that's not the intended way of using Vale. Instead, the Vale documentation instructs you to git-ignore all "synced" configuration files, and only commit custom configuration files. This allows us to avoid introducing big long lists of offensive terms into the monorepo. Those files would still appear if you ran `vale sync` from within the new `.vale` directory. But that is _not_ necessary to do, not even when developing the Vale configuration, because everything can run from within the built docker image. Running Vale using Docker also allows us to avoid contributors needing to install Vale locally, and avoids potential conflicts in Vale versions by pinning the Vale version we use in the base docker image's tag.

We will not use a published image in any circumstance because: (a) the vale docker image is small and after the first time it is downloaded it is reused every time, unlike needing to redownload the entire published image each time the configuration changes, and (b) it avoids a whole host of complexity in our CI and local linting when changes to the Vale configuration conflict with the previous Vale configuration. For example, if a new configuration conflicts with the currently published configuration, the Vale pre-commit step would fail when running the published image! This was too hard to resolve, and would require looking at the git history compared to `HEAD` in `pre-commit`, or juggling new conditionals in our already complex ci_cd.yml workflow. Using a local image every time avoids all of this complexity without any downsides. In fact, it is positive because the base vale image is always reused instead of needing to redownload it any time the Vale configuration changes (excluding updates to the version of the base image).

Run vale using `just .vale/run`. This also builds the docker image. Refer to `.vale/README.md` and other new in-code documentation for rationale behind various configuration choices.

The pre-commit hook also excludes project proposals and changelogs. Changelogs are automatically generated, and as such, I don't think they should be linted for editorial content. For example, if a PR title doesn't pass Vale, I don't think it's accurate to change the title in the changelog. That's my personal judgement, but if other folks want to run Vale there, I would also think that's fine and wouldn't push against it.

The project proposals, on the other hand, are complex. They raise a lot of errors in Vale and are the bulk of our documentation, line-by-line. They are also not "living" documents, that we make changes to over time. Once they're in, they're more or less meant to stay as-is, with changes to the project reflected in the project threads (that's my understanding). Going back and making retroactive editorial changes to those files doesn't feel right to me. On the other hand, if folks _do_ want to do that, then I also wouldn't push back here, I'd just ask that it gets done in a separate PR.

This configuration is just a starting point, and we can incorporate additional rules in the future, if we like, even from other style guides. Red Hat's guide in particular has a lot of compelling options, including scentence complexity checks and a much better passive voice check than the other ones I've seen.

## Testing Instructions

Run `just .vale/run`. Make changes that will cause issues with Vale and confirm they raise errors as you expect. Confirm no unnecessary errors are present.

Read over the Vale configuration and through the Vale documentation to confirm that it all makes sense and is appropriate for our project.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
An alternative to using a published image is to run the vale linting step using `just .vale/ci` that builds the image and then runs it with the configuration we need. That would make it easier to test the CI configuration locally too, without needing to duplicate the CLI args between pre-commit and the justfile.

File: .github/filters.yml
Comment (krysal) [inline .github/filters.yml:73]: Does it need its own section? Why not just keep it under "lint"?

File: justfile
Comment (krysal) [inline justfile:93]: Can this be broken down at least into two lines to make it more readable?
```suggestion
    python3 pre-commit.pyz run {{ hook }} \
    {{ if files =~ "all-files" { "" } else if files == "" { "--all-files" } else { "--files" } }}  {{ files }}
```

**Review by krysal — COMMENTED** [review]
This looks pretty cool! I haven't read everything, so for now, I left a few questions.

File: .github/filters.yml
Comment (sarayourfriend) [reply .github/filters.yml:73]: Totally, I'll move it to `lint` :+1:

File: justfile
Comment (sarayourfriend) [reply justfile:93]: I can remove this change entirely now.

File: .github/filters.yml
Comment (sarayourfriend) [reply .github/filters.yml:73]: Ah, actually, I remember now why I had this separated in addition to under `lint`: the `get-changes` CI step uses this filters.yml to determine the groups of changes, and we need to detect changes in `.vale` to build and publish the vale image when the vale configuration changes. This is what makes the changes in `set_matrix_images.py` actually cause the docker image to get built (and published), because that script checks for `vale` in the set of changes:

https://github.com//WordPress/openverse/blob/b6ccc486ebd561f9777ee5128948e6b2e262c77b/automations/python/workflows/set_matrix_images.py#L61

Comment (sarayourfriend) [reply .github/filters.yml]: I've removed this altogether in favour of a much simpler local-docker image approach. I updated the PR description with details and rationale.

**Review by stacimc — APPROVED** [review]
~Very~ Exceptionally cool :) Documentation improvements all look good to me as well; this is a welcome change!

I agree with you re: project proposals and especially the changelogs.

One thing I noticed when testing is that Vale appears to ignore anything in code blocks using the backticks. In our documentation some of these are actual code blocks but others are notes/warnings that probably ought to be checked. For example if I add this in a doc file, it won't raise any warnings or errors:

> \`\`\`{warning}
This note is very likely to to cause Vale errors.
\`\`\`

Not a blocker but perhaps something to investigate in a follow up? I looked briefly and was unable to determine if this is a bug or intentional behavior by Vale.

**Review by krysal — APPROVED** [review]
Nice addition! The last thing I think is adding the `.vale/` folder to the CODEOWNERS. 
Thank you for this comprehensive setup.

File: catalog/dags/providers/provider_api_scripts/inaturalist.py
Comment (sarayourfriend) [reply catalog/dags/providers/provider_api_scripts/inaturalist.py:10]: Yes, that is precisely why. Sorry, I should have left a note regarding this particular change.

**Comment by sarayourfriend** [comment]
Okay, I've updated the PR with the following:

1. Check Markdown files outside of `documentation`. As part of this I cleaned up the glob configuration in `.vale/justfile` to make it slightly easier to read and maintain (though not by much, due to limitations in the glob syntax and just/scripting while avoiding too much complexity)
2. Check MDX files as Markdown
3. Check admonitions
4. Fix a lot of false positives by updating `Openverse.TermCasing` to ignore path-like phrases

**Comment by sarayourfriend** [comment]
@obulat @dhruvkb I think this PR needs a review from one of y'all to be able to get merged? I can't understand why else the existing 2 approvals wouldn't be sufficient.

**Review by dhruvkb — APPROVED** [review]
I think that's because "frontend" owns at least one of the modified files and Staci isn't a member of that group.

**Pull request merged**
